examples/mpg_example.py

```diff
@@ -0,0 +1,308 @@
+         os.path.join(working_dir, transform_fn_io.TRANSFORM_FN_DIR),
+         raw_features))
+
+    return tf.estimator.export.ServingInputReceiver(transformed_features,
+                                                    default_inputs)
+
+  return serving_input_fn
+
+
+def train_and_evaluate(working_dir, num_train_instances=NUM_TRAIN_INSTANCES,
+                       num_test_instances=NUM_TEST_INSTANCES):
+  """Train the model on training data and evaluate on eval data.
+
+  Args:
+    working_dir: Directory to read transformed data and metadata from and to
+        write exported model to.
+    num_train_instances: Number of instances in train set
+    num_test_instances: Number of instances in test set
+
+  Returns:
+  """
+
+  one_hot_columns = [
+    tf.feature_column.indicator_column(
+       tf.feature_column.categorical_column_with_identity(key=key,
+                                                          num_buckets=num_buckets))
+    for key, num_buckets in zip(CATEGORICAL_FEATURE_KEYS, BUCKET_SIZES)]
+
+  real_valued_columns = [tf.feature_column.numeric_column(key, shape=())
+                         for key in NUMERIC_FEATURE_KEYS]
+
+  estimator = tf.estimator.DNNRegressor(
+     feature_columns=real_valued_columns + one_hot_columns,
+     model_dir=os.path.join(working_dir, "logs_directory"),
+     optimizer=tf.train.AdamOptimizer(),
+     hidden_units=[10, 5])
+
+  train_spec = tf.estimator.TrainSpec(
+     input_fn=lambda: _make_training_input_fn(working_dir, "auto-mpg.csv",
+                                              BATCH_SIZE),
+     max_steps=TRAIN_NUM_EPOCHS * num_train_instances / BATCH_SIZE)
+
+  eval_spec = tf.estimator.EvalSpec(
+     input_fn=lambda: _make_training_input_fn(working_dir, "auto-mpg-test.csv",
+                                              BATCH_SIZE),
+     throttle_secs=10, steps=num_test_instances / BATCH_SIZE)
+
+  tf.estimator.train_and_evaluate(estimator, train_spec, eval_spec)
+
+  # Export the model.
+  serving_input_fn = _make_serving_input_fn(working_dir)
+  exported_model_dir = os.path.join(working_dir, EXPORTED_MODEL_DIR)
+  estimator.export_savedmodel(exported_model_dir, serving_input_fn)
+
+
+def main():
+  parser = argparse.ArgumentParser()
+  parser.add_argument(
+     'input_data_dir',
+     help='path to directory containing input data')
+  parser.add_argument(
+     '--working_dir',
+     help='optional, path to directory to hold transformed data')
+  args = parser.parse_args()
+
+  if args.working_dir:
+    working_dir = args.working_dir
+  else:
+    working_dir = tempfile.mkdtemp(dir=args.input_data_dir)
+
+  train_data_file = os.path.join(args.input_data_dir, 'auto-mpg.csv')
+
+  # Will write a SavedModel and metadata to two subdirectories of
+  # working_dir, given by transform_fn_io.TRANSFORM_FN_DIR and
+  # transform_fn_io.TRANSFORMED_METADATA_DIR respectively.
+  create_transform_fn(train_data_file, working_dir)
```

Comment:
This fails when running in clustered mode. I don't know how to tell tensorflow that this needs only to be done once (by any of the nodes)

Reply:
What kind of cluster are you using?  Do the other examples work in clustered mode?

Reply:
I'm referring to Google Cloud Machine Learning Engine. When using train_and_evaluate you can run it there with a "scaleTier: STANDARD_1" (clustered) without any change in the code. Other examples I guess won't work (don't use train_and_evaluate). The example I'm providing works (provided I copy the transform_fn manually to the gs bucket, or I remove all tf-transform step altogether)

Reply:
In my mind transformation 'on-the-fly' makes sense when the data set is huge, in which case you'll probably want to use a cluster as well. That's why I thought the example would provide more value if combining both features.

Reply:
What is we set a "mode" command line argument that took values "analyze_and_train" (default), "analyze" or "train"?  Would this allow the cloud trainer job to just transform?  Note that we don't plan to provide instructions for running Cloud ML Engine here, but ideally these examples would run on Cloud ML Engine without modification.

Reply:
The gcloud command line doesn't seem to support trainer parameters when using the gcloud command line (at first sight). Anyway I just wanted to raise awareness of this possible issue (how tf-transform fits with Cloud ML Engine).
Maybe we could just ignore it for the sake of this simple example, for the time being.

Reply:
Sounds good, there are a variety of samples that show usage with Cloud ML Engine, and these are written with separate scripts.  This example is a single script but could easily be split into a transform script and a train script.